lib/proxy_api/template.rb

```diff
@@ -11,12 +11,12 @@ def template_url
         return response["templateServer"]
       end
     rescue SocketError, Timeout::Error, Errno::EINVAL, Errno::ECONNRESET,
```

Comment:
This code was commited as part of the 2094e4e8b049e and this concern should have been discussed there, but okay. I can rethrow the exception with a nicer error message.

The issue here is this is used in various contexts (template rendering, bootdisk rake tasks) and bare error message like "Connection refused!" says nothing. This is what I am trying to improve.